webapp/src/simulator.ts

```diff
@@ -66,6 +69,28 @@ export function init(root: HTMLElement, cfg: SimulatorConfig) {
         }
     }
 
+    // Map simulator extensions from approved repos
+    const simulatorExtensions: pxt.Map<pxt.SimulatorExtensionConfig> = {};
+    Object.entries(trgConfig?.packages?.approvedRepoLib || {})
+        .map(([k, v]) => ({ k: k, v: v.simx }))
+        .filter(e => !!e.v)
+        .forEach(e => simulatorExtensions[e.k] = {
+            index: "index.html", // default to index.html
+            aspectRatio: pxt.appTarget.simulator.aspectRatio || 1.22, // fallback to 1.22
+            permanent: true, // default to true
+            ...e.v
+        });
+    // Add in test simulator extensions
+    Object.entries(pxt.appTarget?.simulator?.testSimulatorExtensions || {})
+        .map(([k, v]) => ({ k: k, v: v as pxt.SimulatorExtensionConfig }))
+        .filter(e => !!e.v)
+        .forEach(e => simulatorExtensions[e.k] = {
+            index: "index.html", // default to index.html
+            aspectRatio: pxt.appTarget.simulator.aspectRatio || 1.22, // fallback to 1.22
+            permanent: true, // default to true
+            ...e.v
+        });
+
```

Comment:
Is it okay to do this inline replacement for these? If something bad happens is there a way to recover?

Reply:
It is correct behavior for test registrations replace existing entries because we may be testing a new version of an existing extension. We must never ship extension registrations in the `testSimulatorExtensions` field. It is only for making upload targets and should always be empty in PRs.